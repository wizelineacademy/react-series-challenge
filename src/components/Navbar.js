import React, { Component } from "react";
import { NavLink, Link } from "react-router-dom";
import {UlStyled, LiStyled, InputTextStyled,ButtonTextStyled} from "./Navbar.style"

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tag: "nothing"
    };
  }
  render() {
    return (
      <React.Fragment>
        <UlStyled>
          <LiStyled>
            <NavLink to="/">Home</NavLink>
          </LiStyled>
          <LiStyled>
          <NavLink to="/fav">Favs</NavLink>
          </LiStyled>
          <LiStyled>
          <InputTextStyled
              type="text"
              name="search"
              onChange={event => { this.setState({ tag: event.target.value }); }}
            />
          </LiStyled>
          <LiStyled>
            <Link
              to={{
                pathname: "/search",
                state: { query: this.state.tag }
              }}
            >
              <ButtonTextStyled type="button" id="add">Search</ButtonTextStyled>
            </Link>
          </LiStyled>
        </UlStyled>
      </React.Fragment>
    );
  }
}

export default Navbar;