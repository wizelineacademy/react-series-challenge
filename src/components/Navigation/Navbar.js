import React from "react";
import { withRouter } from "react-router-dom"
import SearchBar from "../SearchBar/SearchBar";
import { ContainerNavbarStyled, NavbarLinkStyle } from "./NavbarStyle";

const Navbar = ({ location }) => (
  // location.pathame
  <ContainerNavbarStyled>
    <div>
      <NavbarLinkStyle exact to="/">Home</NavbarLinkStyle>
      <NavbarLinkStyle to="/favorites">Favoritos</NavbarLinkStyle>
    </div>
    <SearchBar from={location.pathname}/>
  </ContainerNavbarStyled>
);

export default withRouter(Navbar);
