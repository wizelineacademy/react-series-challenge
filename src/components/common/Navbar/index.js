import React from 'react'
import NavbarStyled from './NavbarStyled'
import Link from '../Link'



const Navbar = (props)=>{
    return(
        <NavbarStyled>
                <Link to="/">Home</Link>
                <Link to="/favorites">Favorites</Link>
        </NavbarStyled>
    )
}

export default Navbar