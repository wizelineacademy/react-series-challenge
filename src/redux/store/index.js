import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../reducers';
import customMiddleware from '../middleware';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../sagas';

const sagaMiddleware = createSagaMiddleware();

const initialState = {};
const enhancers = [];
const middleware = [
  sagaMiddleware,
  ...customMiddleware,
];

const composedEnhancers = compose(
  applyMiddleware(...middleware),
  ...enhancers,
);

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers,
);

sagaMiddleware.run(rootSaga);

export default store;
