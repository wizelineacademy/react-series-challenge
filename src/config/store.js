import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';
import {APPLICATION_LOCAL_STORAGE_KEY} from './config';
import {initialState as gifInitialState} from '../reducers/gif';

// Middlewares
const sagaMiddleware = createSagaMiddleware();
const middlewares = [
  sagaMiddleware
];

// Enhancers
const createEnhancers = () => {
  let enhancers = [];

  /**
   * Uncomment this lines and add the redux-devtools package to enable redux dev
   *  tools into development mode.
   */
  // if (process.env.NODE_ENV !== 'production') {
  //   enhancers.push(
  //     window.__REDUX_DEVTOOLS_EXTENSION__
  //       && window.__REDUX_DEVTOOLS_EXTENSION__());
  // }

  return enhancers;
};

const persistStateDetails = (state) => {
  try {
    let stateCopy = JSON.parse(JSON.stringify(state));

    // Remove unnecesary state details
    stateCopy.gifs.fetching = gifInitialState.fetching;
    stateCopy.gifs.displayed = gifInitialState.displayed;
    stateCopy.gifs.favoritesTextFilter = gifInitialState.favoritesTextFilter;

    let serializedState = JSON.stringify(stateCopy);
    localStorage.setItem(APPLICATION_LOCAL_STORAGE_KEY, serializedState);
  } catch (e) {}
}

const retrievePreviousStateDetails = () => {
  let result = {};

    try {
      let serializedState =
        window.localStorage.getItem(APPLICATION_LOCAL_STORAGE_KEY);

      if (serializedState) {
        result = JSON.parse(serializedState);
      }
    } catch (e) {}

  return result;
}

const enhancers = createEnhancers();
const composedEnhancers = (enhancers.length > 0)
  ? compose(applyMiddleware(...middlewares), ...enhancers)
  : applyMiddleware(...middlewares);


const initializeStore = (initialState) => {
  if (!initialState) {
    // Rehydrate the store
    initialState = retrievePreviousStateDetails();
  }

  // Store object
  const store = createStore(rootReducer, initialState, composedEnhancers);

  store.subscribe(() => persistStateDetails(store.getState()));

  sagaMiddleware.run(rootSaga);

  return store;
}

export default initializeStore;
