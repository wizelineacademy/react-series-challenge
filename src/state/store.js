import createSagaMiddleware from "redux-saga";
import { createStore, applyMiddleware } from "redux";

import rootSaga from "./sagas";
import reducers from "./reducers";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(reducers, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

export default store;
