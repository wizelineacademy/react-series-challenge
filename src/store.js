import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import { gif, trending } from './reducers';

const reducer = combineReducers({ gif, trending });

export default createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));
