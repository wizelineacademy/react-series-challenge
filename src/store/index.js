import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleWare from 'redux-saga';
import rootReducer from '../reducers';
import logger from '../middlewares/logger';
import rootSaga from '../sagas';

const sagaMiddleware = createSagaMiddleWare();

// const initialState = {};
const middlewares = [
  logger,
  sagaMiddleware
];

const appliedMiddleware = applyMiddleware(...middlewares);

const store = createStore(
  rootReducer,
  // initialState,
  appliedMiddleware
);

sagaMiddleware.run(rootSaga);

export default store;