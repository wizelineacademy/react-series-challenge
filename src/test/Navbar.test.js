import React from 'react';
import { shallow, mount } from 'enzyme';
import Navbar from '../components/navbar';
import store from '../store'

describe('test Navbar Render', () => {
  it('render structure', () => {
    const renderedComponent = shallow(<Navbar store={store} />);
    expect(renderedComponent).toMatchSnapshot();
  })
})