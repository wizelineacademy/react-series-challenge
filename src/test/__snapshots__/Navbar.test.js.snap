// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navbar Render render structure 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.li>
          <Link
                    replace={false}
                    to="/"
          >
                    Home
          </Link>
</styled.li>,
        <styled.li>
          <Link
                    replace={false}
                    to="/favorites"
          >
                    Favorites
          </Link>
</styled.li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
>
            Home
</Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            replace={false}
            to="/favorites"
>
            Favorites
</Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Favorites",
            "replace": false,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.li>
            <Link
                        replace={false}
                        to="/"
            >
                        Home
            </Link>
</styled.li>,
          <styled.li>
            <Link
                        replace={false}
                        to="/favorites"
            >
                        Favorites
            </Link>
</styled.li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              Home
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/favorites"
>
              Favorites
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Favorites",
              "replace": false,
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
